mlflow/models/evaluation/base.py

```diff
@@ -151,6 +156,161 @@ def __str__(self):
         return "EvaluationMetric(" + ", ".join(parts) + ")"
 
 
+def dynamically_generate_genai_eval_metric(eval_fn, with_llm_judge=False):
```

Comment:
can you insert a comment explaining why we need to do this?

Reply:
Added!